centreon/src/Core/HostGroup/Infrastructure/Repository/DbReadHostGroupRepository.php

```diff
@@ -306,6 +306,18 @@ private function getFindHostGroupConcatenator(array $accessGroupIds = []): SqlCo
                         `:db`.`hostgroup` hg
                     SQL
             )
+            ->appendJoins(
+                <<<'SQL'
+                    LEFT JOIN `:db`.hostgroup_relation hgr
+                        ON hg.hg_id = hgr.hostgroup_hg_id
+                    LEFT JOIN `:db`.host h
+                        ON hgr.host_host_id = h.host_id
+                    LEFT JOIN `:db`.hostcategories_relation hcr
+                        ON h.host_id = hcr.host_host_id
+                    LEFT JOIN `:db`.hostcategories hc
+                        ON hcr.hostcategories_hc_id = hc.hc_id
+                    SQL
+            )
```

Comment:
Is this necessary if the search parameter host_category_ids is not present ? 
If not add a check for this parameter before doing the appendJoin